### PR TITLE
Photo ui

### PR DESCRIPTION
**Pull request by Br3nda**
- Editing a photo's title
- Don't overwrite photo's title if we re-import from flickr later
- New functionality, we can now remove a link from an item (harvest, garden, seed, plantings). so we don't need to delete the whole photo anymore if we link to the wrong item.
- Places the photo at the top, as that is what the user is seeking on this page.
- Makes the photo size responsive, so it fits in phone screens
- Fixed bug that was pushing the menu off screen on phones

### CODE REVIEW

**Comment by Br3nda** [comment]
before
image


after
image

**Comment by Br3nda** [comment]
before
image

after
image

**Comment by Br3nda** [comment]
and when not signed in, it will look like this
image

image

File: app/models/photo.rb [hunk 1]
```diff
@@ -10,6 +10,10 @@ class Photo < ActiveRecord::Base
 
   default_scope { order("created_at desc") }
 
+  def associations?
+    plantings.any? || harvests.any? || gardens.any? || seeds.any?
```
Comment (Br3nda) [inline app/models/photo.rb]: should probably add counter caches for these

**Comment by Br3nda** [comment]
fixes #474

File: app/views/photos/_photo_association_delete.haml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- if can? :edit, photo
+  = link_to photo_associations_path(photo_id: photo.id, type: type, id: thing.id),
+    method: 'delete', class: 'btn btn-default btn-xs' do
+    %span.glyphicon.glyphicon-remove{ title: "Remove link" }
```
Comment (CloCkWeRX) [inline app/views/photos/_photo_association_delete.haml]: Need i18n?

File: app/views/photos/show.html.haml [hunk 3]
```diff
@@ -8,7 +8,19 @@
...
 .row
-  .col-md-6
+  .col-md-8
+    %p= image_tag(@photo.fullsize_url, alt: @photo.title, class: 'img img-responsive')
+
+  .col-md-4
+    %p
+      - if can? :destroy, @photo
+        = link_to @photo, method: :delete,
+          data: { confirm: 'Are you sure?' }, class: 'btn btn-default btn-xs' do
```
Comment (CloCkWeRX) [inline app/views/photos/show.html.haml]: Ditto i18n

**Comment by CloCkWeRX** [comment]
Will let you decide if you want to internationalise bits or just do that later.

**Pull request merged**